src/java/org/apache/cassandra/index/sai/plan/Expression.java

```diff
@@ -335,6 +344,18 @@ private boolean termMatches(ByteBuffer term, ByteBuffer requestedValue)
             case RANGE:
                 isMatch = isLowerSatisfiedBy(term) && isUpperSatisfiedBy(term);
                 break;
+
+            case IN:
+                ListType<?> type = ListType.getInstance(indexTermType.columnMetadata().type, true);
+                List<? extends ByteBuffer> buffers = type.unpack(requestedValue);
+                for (ByteBuffer value : buffers)
+                {
+                    if (indexTermType.compare(term, value) == 0)
+                    {
+                        isMatch = true;
+                        return true;
+                    }
+                }
```

Comment:
nit: Just to future-proof (in case we add another operator), I'd throw a `break;` after the body of the `case IN:` here.